src/index.js

```diff
@@ -56,7 +56,7 @@ const cloneWithChildren = (node, children, isRootEl, level) => {
 };
 
 const validateTree = (node) => {
-  if (typeof node === 'string' || isAtomComponent(node)) {
+  if (['string', 'number'].includes(typeof node) || isAtomComponent(node)) {
```

Comment:
I realized we need to check for other values here. let's add a null/undefined check as the first thing, like
```js
if (node == null || ...)
```

and at the end, when checking for `node.props.children`, let's be a bit more defensive, and check for `node.props` first, like
```js
if (node.props && node.props.children)
```

Reply:
I've added a test case and checks. I'm not sure if we need to check `null/undefined`.

Reply:
nit: checking for `node == null` will take care of both the null and undefined checks